examples_apps/run_state_learner.py

```diff
@@ -0,0 +1,432 @@
+parameterized Gaussian *and* non-Gaussian gates. **The non-Gaussian
+gates provide both the nonlinearity and the universality of the model.**
+To this end, we employ the CV quantum neural network architecture as described in
+`Killoran et al. <https://journals.aps.org/prresearch/abstract/10.1103/PhysRevResearch.1.033063>`__:
+
+.. figure:: https://i.imgur.com/NEsaVIX.png
+   :alt: layer
+
+Here,
+
+-  :math:`\mathcal{U}_i(\theta_i,\phi_i)` is an N-mode linear optical
+   interferometer composed of two-mode beamsplitters
+   :math:`BS(\theta,\phi)` and single-mode rotation gates
+   :math:`R(\phi)=e^{i\phi\hat{n}}`,
+
+-  :math:`\mathcal{D}(\alpha_i)` are single mode displacements in the
+   phase space by complex value :math:`\alpha_i`,
+
+-  :math:`\mathcal{S}(r_i, \phi_i)` are single mode squeezing operations
+   of magnitude :math:`r_i` and phase :math:`\phi_i`, and
+
+-  :math:`\Phi(\lambda_i)` is a single mode non-Gaussian operation, in
+   this case chosen to be the Kerr interaction
+   :math:`\mathcal{K}(\kappa_i)=e^{i\kappa_i\hat{n}^2}` of strength
+   :math:`\kappa_i`.
+
+
+Hyperparameters
+---------------
+
+First, we must define the **hyperparameters** of our layer structure:
+
+-  ``cutoff``: the simulation Fock space truncation we will use in the
+   optimization. The TensorFlow backend will perform numerical
+   operations in this truncated Fock space when performing the
+   optimization.
+
+-  ``depth``: The number of layer ansatz in our variational quantum
+   circuit. As a general rule, increasing the number of layers (and
+   thus, the number of parameters we are optimizing over) increases the
+   optimizers chance of finding a reasonable local minimum in the
+   optimization landscape.
+
+-  ``reps``: the number of steps in the optimization routine performing
+   gradient descent
+
+Some other optional hyperparameters include:
+
+-  The standard deviation of initial parameters. Note that we make a
+   distinction between the standard deviation of *passive* parameters
+   (those that preserve photon number when changed, such as phase
+   parameters), and *active* parameters (those that introduce or remove
+   energy from the system when changed).
+"""
+
+import strawberryfields as sf
+from strawberryfields.ops import *
+from strawberryfields.utils import operation
+
+# Cutoff dimension
+cutoff = 9
+
+# Number of layers
+depth = 15
+
+# Number of steps in optimization routine performing gradient descent
+reps = 200
+
+# Learning rate
+lr = 0.05
+
+# Standard deviation of initial parameters
+passive_sd = 0.1
+active_sd = 0.001
+
+######################################################################
+# The layer parameters :math:`\vec{\theta}`
+# -----------------------------------------
+#
+# We use TensorFlow to create the variables corresponding to the gate
+# parameters. Note that each variable has shape ``[depth]``, with each
+# individual element representing the gate parameter in layer :math:`i`.
```

Comment:
```suggestion
# parameters. Note that we focus on a single mode circuit where
# each variable has shape ``[depth]``, with each
# individual element representing the gate parameter in layer :math:`i`.
```
I was confused at first why this was the shape

Reply:
(Minor suggestion): how about having a tuple based shape representation for `depth`: ``(depth,)``? (gets converted internally to a tuple anyways, just thought otherwise questions might be raised why it is a `list` here)